contracts/axelarnet-gateway/src/clients/external.rs

```diff
@@ -11,36 +12,35 @@ pub struct AxelarExecutableMsg {
     pub payload: HexBinary,
 }
 
-/// Crate-specific `ExecuteMsg` type wraps the `AxelarExecutableMsg` for the AxelarExecutable client.
-#[cw_serde]
-pub(crate) enum AxelarExecutableExecuteMsg {
-    /// Execute the message at the destination contract with the corresponding payload.
-    Execute(AxelarExecutableMsg),
+pub struct PayloadExecutor<'a> {
```

Comment:
```suggestion
pub struct AxelarExecutableClient<'a> {
```
I'd prefer to still call this AxelarExecutable client to match the terminology used by external gateways. E.g. https://github.com/axelarnetwork/axelar-gmp-sdk-solidity/blob/main/contracts/interfaces/IAxelarGMPExecutable.sol

Makes it easier for devrel/docs since people are used to the EVM side.

Reply:
since we're not calling it `IAxelarGMPExecutable` I think it's better to have a name that actually means something 😉

Reply:
I renamed it though, let me know what you think

Reply:
Linked to the another one, here's the [AxelarExecutable](https://github.com/axelarnetwork/axelar-gmp-sdk-solidity/blob/main/contracts/executable/AxelarExecutable.sol#L8). `Ixyz` is just the solidity convention for an interface.

Similar to `CallContract`, I prefer reusing existing terminology for GMP, instead of using new ones, and `CrossChainExecutor` isn't really much better in my mind (is it executing on a remote chain or locally, is it some app built on top of GMP etc.)

Reply:
this client should not be exposed anyway. I ended up calling it `external::Client`. I believe that's clear enough